python/ray/autoscaler/_private/aws/cloudwatch/cloudwatch_helper.py

```diff
@@ -0,0 +1,642 @@
+            else default_instance_profile_name
+
+    @staticmethod
+    def resolve_iam_role_name(config: Dict[str, Any],
+                              default_iam_role_name: str) -> str:
+        """Get default cloudwatch iam role name.
+
+        Args:
+            config: provider section of cluster config file.
+            default_iam_role_name: default ray iam role name.
+
+        Returns:
+            default cloudwatch iam role name if cloudwatch config file exists.
+            default ray iam role name if cloudwatch config file doesn't exist.
+        """
+        cwa_cfg_exists = CloudwatchHelper.cloudwatch_config_exists(
+            config, "config")
+        return CLOUDWATCH_RAY_IAM_ROLE if cwa_cfg_exists \
+            else default_iam_role_name
+
+    @staticmethod
+    def resolve_policy_arns(config: Dict[str, Any], iam: Any,
+                            default_policy_arns: List[str]) -> List[str]:
+        """Attach necessary AWS policies for CloudWatch related operations.
+
+        Args:
+            config: provider section of cluster config file.
+            iam: AWS iam resource.
+            default_policy_arns: List of default ray AWS policies.
+
+        Returns:
+            list of policy arns including additional policies for CloudWatch
+                related operations if cloudwatch agent config is specifed in
+                cluster config file.
+        """
+        cwa_cfg_exists = CloudwatchHelper.cloudwatch_config_exists(
+            config, "config")
+        if cwa_cfg_exists:
+            cloudwatch_managed_policy = {
+                "Version": "2012-10-17",
+                "Statement": [{
+                    "Effect": "Allow",
+                    "Action": [
+                        "ssm:SendCommand", "ssm:ListCommandInvocations",
+                        "iam:PassRole"
+                    ],
+                    "Resource": "*"
+                }]
+            }
+            iam_client = iam.meta.client
+            iam_client.create_policy(
+                PolicyName="CloudwatchManagedPolicies",
+                PolicyDocument=json.dumps(cloudwatch_managed_policy))
+            sts_client = client_cache("sts", config["region"])
+            account_id = sts_client.get_caller_identity().get("Account")
+            managed_policy_arn = \
+                "arn:aws:iam::{}:policy/CloudwatchManagedPolicies".\
+                format(account_id)
+            policy_waiter = iam_client.get_waiter("policy_exists")
+            policy_waiter.wait(
+                PolicyArn=managed_policy_arn,
+                WaiterConfig={
+                    "Delay": 2,
+                    "MaxAttempts": 200
+                })
+            default_policy_arns.extend([
+                "arn:aws:iam::aws:policy/CloudWatchAgentAdminPolicy",
+                "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore",
+                managed_policy_arn
+            ])
+        return default_policy_arns
```

Comment:
```suggestion
            new_policy_arns = copy.copy(default_policy_arns)
            new_policy_arns.extend([
                "arn:aws:iam::aws:policy/CloudWatchAgentAdminPolicy",
                "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore",
                managed_policy_arn
            ])
            return new_policy_arns
```
Minor: Since this isn't a high-performance code path and we're typically dealing with small lists, returning a new copy of the input policy ARNs allows us to more easily avoid unintentional side-effects, and do things like diff the old vs. new policy ARNs later.